app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/WebViewSingleton.java

```diff
@@ -0,0 +1,275 @@
+            LOG.debug("WEBVIEW uuid changed, restarting");
+            currentRunningUUID = uuid;
+            new Handler(webViewSingleton.mainLooper).post(new Runnable() {
+                @Override
+                public void run() {
+                    webViewSingleton.instance.onResume();
+                    webViewSingleton.instance.removeJavascriptInterface("GBjs");
+                    webViewSingleton.instance.addJavascriptInterface(jsInterface, "GBjs");
+                    webViewSingleton.instance.loadUrl("file:///android_asset/app_config/configure.html?rand=" + Math.random() * 500);
+                }
+            });
+            if (!internetHelperBound) {
+                Intent intent = new Intent();
+                intent.setComponent(new ComponentName("nodomain.freeyourgadget.internethelper", "nodomain.freeyourgadget.internethelper.MyService"));
+                webViewSingleton.contextWrapper.getApplicationContext().bindService(intent, internetHelperConnection, Context.BIND_AUTO_CREATE);
+            }
+        }
+
+    }
+
+    public static void appMessage(GBDeviceEventAppMessage message) {
+
+        final String jsEvent;
+        if (webViewSingleton.instance == null) {
+            LOG.warn("WEBVIEW is not initialized, cannot send appMessages to it");
+            return;
+        }
+
+        if (!message.appUUID.equals(currentRunningUUID)) {
+            LOG.info("WEBVIEW ignoring message for app that is not currently running: " + message.appUUID + " message: " + message.message + " type: " + message.type);
+            return;
+        }
+
+        // TODO: handle ACK and NACK types with ids
+        if (message.type != GBDeviceEventAppMessage.TYPE_APPMESSAGE) {
+            jsEvent = (GBDeviceEventAppMessage.TYPE_NACK == GBDeviceEventAppMessage.TYPE_APPMESSAGE) ? "NACK" + message.id : "ACK" + message.id;
+            LOG.debug("WEBVIEW received ACK/NACK:" + message.message + " for uuid: " + message.appUUID + " ID: " + message.id);
+        } else {
+            jsEvent = "appmessage";
+        }
+
+        final String appMessage = parseIncomingAppMessage(message.message, message.appUUID);
+        LOG.debug("to WEBVIEW: event: " + jsEvent + " message: " + appMessage);
+        new Handler(webViewSingleton.mainLooper).post(new Runnable() {
+            @Override
+            public void run() {
+                webViewSingleton.instance.evaluateJavascript("Pebble.evaluate('" + jsEvent + "',[" + appMessage + "]);", new ValueCallback<String>() {
+                    @Override
+                    public void onReceiveValue(String s) {
+                        //TODO: the message should be acked here instead of in PebbleIoThread
+                        LOG.debug("Callback from appmessage: " + s);
+                    }
+                });
+            }
+        });
+    }
+
+    public static void disposeWebView() {
+        if (internetHelperBound) {
```

Comment:
Does re-setting up the whole webview work after it has been disposed once?

Reply:
It works without issues on my device after multiple disconnects.